test/rhsmlib/dbus/test_register.py

```diff
@@ -321,257 +312,159 @@
-
-        def assertions(*args):
-            result = args[0]
-            self.assertRegex(result, r"/run/dbus.*")
-
-        self.dbus_request(assertions, self.interface.Start, dbus_method_args)
-
-    def test_same_socket_on_subsequent_opens(self):
-        dbus_method_args = [""]
-
-        def assertions(*args):
-            # Assign the result as an attribute to this function.
-            # See http://stackoverflow.com/a/27910553/6124862
-            assertions.result = args[0]
-            self.assertRegex(assertions.result, r"/run/dbus.*")
-
-        self.dbus_request(assertions, self.interface.Start, dbus_method_args)
+class RegisterDBusObjectTest(DBusServerStubProvider):
+    dbus_class = RegisterDBusObject
+    dbus_class_kwargs = {}
 
-        # Reset the handler_complete_event so we'll block for the second
-        # dbus_request
-        self.handler_complete_event.clear()
+    def tearDown(self) -> None:
+        """Make sure the domain server is stopped once the test ends."""
+        with contextlib.suppress(rhsmlib.dbus.exceptions.Failed):
+            self.obj.Stop.__wrapped__(self.obj, "en_US.UTF-8")
 
-        def assertions2(*args):
-            result2 = args[0]
-            self.assertEqual(assertions.result, result2)
+    def test_Start(self):
+        substring = "/run/dbus.*"
+        result = self.obj.Start.__wrapped__(self.obj, "en_US.UTF-8")
+        self.assertRegex(result, substring)
 
-        self.dbus_request(assertions2, self.interface.Start, dbus_method_args)
+    def test_Start__two_starts(self):
+        """Test that opening the server twice returns the same address"""
+        result_1 = self.obj.Start.__wrapped__(self.obj, "en_US.UTF-8")
+        result_2 = self.obj.Start.__wrapped__(self.obj, "en_US.UTF-8")
+        self.assertEqual(result_1, result_2)
 
-    def test_cannot_close_what_is_not_opened(self):
-        dbus_method_args = [""]
-        with self.assertRaises(dbus.exceptions.DBusException):
-            self.dbus_request(None, self.interface.Stop, dbus_method_args)
-
-    def test_closes_domain_socket(self):
-        dbus_method_args = [""]
-
-        def get_address(*args):
-            address = args[0]
-            _prefix, _equal, address = address.partition("=")
-            get_address.address, _equal, _suffix = address.partition(",")
-
-        self.dbus_request(get_address, self.interface.Start, dbus_method_args)
-        self.handler_complete_event.clear()
+    def test_Start__can_connect(self):
+        result = self.obj.Start.__wrapped__(self.obj, "en_US.UTF-8")
+        prefix, _, data = result.partition("=")
+        address, _, guid = data.partition(",")
 
         sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
         try:
-            # The socket returned for connection is an abstract socket so we have
-            # to begin the name with a NUL byte to get into that namespace.  See
-            # http://blog.eduardofleury.com/archives/2007/09/13
```

Comment:
this comment is valid, as it is not immediate to people not familiar with (Linux) sockets that a UNIX socket with a NULL byte at the beginning of the path is an abstract Linux socket; I'd leave it there